Jenkinsfile

```diff
@@ -27,41 +27,45 @@ def getChangeDetail() {
     return changeDetail
 }
 
-def sendEmailNotification() {    
-
-    script{ 
-        final def EXTRECIPIENTS = emailextrecipients([
-            [$class: 'CulpritsRecipientProvider'],
-            [$class: 'RequesterRecipientProvider'],
-            [$class: 'DevelopersRecipientProvider']
-        ])
+def IS_MASTER_BRANCH = env.BRANCH_NAME == "master"
+def IS_RELEASE_BRANCH = (env.BRANCH_NAME ==~ /\d+\.\d+\.\d+/)
 
-        def RECIPIENTS = EXTRECIPIENTS != null ? EXTRECIPIENTS : env.CHANGE_AUTHOR_EMAIL;
-        def SUBJECT = "${env.JOB_NAME} - Build #${env.BUILD_NUMBER} - ${currentBuild.currentResult}!"
-        def CHANGE = getChangeDetail();
-        
-        emailext(
-            to: RECIPIENTS,
-            subject: "${SUBJECT}",
-            body: """
-                <b>Build result:</b> 
-                <br><br>
-                ${currentBuild.currentResult}
-                <br><br>
-                <b>Project name - Build number:</b>
-                <br><br>
-                ${currentBuild.fullProjectName} - Build #${env.BUILD_NUMBER}
-                <br><br>
-                <b>Check console output for more detail:</b> 
-                <br><br>
-                <a href="${currentBuild.absoluteUrl}console">${currentBuild.absoluteUrl}console</a>
-                <br><br>
-                <b>Changes:</b> 
-                <br><br>
-                ${CHANGE}
-                <br><br>
-            """
-        );
+def sendEmailNotification() {    
+    script { 
+        if (IS_MASTER_BRANCH || IS_RELEASE_BRANCH)  {
```

Comment:
This is totally a personal preference but could you replace this with:

```
if (!(IS_MASTER_BRANCH || IS_RELEASE_BRANCH) {
  println "Skipping email step since branch condition was not met"
  return
}
```

to save a layer of indentation for the rest of the method

Reply:
Also, maybe Codewind doesn't want this, but I want emails for failed PR builds

Reply:
Revised the code. For emails on the failed PR builds, please refer to the discussion here: https://github.com/eclipse/codewind/pull/2838#pullrequestreview-404854805